packages/frontend/src/hooks/useSpaces.ts

```diff
@@ -32,21 +32,25 @@ export const useSpaces = (
     );
 };
 
-const getSpaceSummaries = async (projectUuid: string) => {
+const getSpaceSummaries = async (
+    projectUuid: string,
+    includePrivateSpaces: boolean,
+) => {
     return lightdashApi<SpaceSummary[]>({
-        url: `/projects/${projectUuid}/spaces`,
+        url: `/projects/${projectUuid}/spaces?includePrivateSpaces=${includePrivateSpaces}`,
         method: 'GET',
         body: undefined,
     });
 };
 
 export const useSpaceSummaries = (
     projectUuid: string,
+    includePrivateSpaces: boolean = false,
     queryOptions?: UseQueryOptions<SpaceSummary[], ApiError>,
 ) => {
     return useQuery<SpaceSummary[], ApiError>(
-        ['projects', projectUuid, 'spaces'],
-        () => getSpaceSummaries(projectUuid),
+        ['projects', projectUuid, 'spaces', includePrivateSpaces],
```

Comment:
This means that everytime you toggle you will make a new request to the backend 
[Screencast from 06-07-23 08:14:47.webm](https://github.com/lightdash/lightdash/assets/1983672/27335c6e-a5b0-413b-a5fc-c1389169533b)

Can't we change this so a flag is returned in the response and filtered in the frontend ?